man/fi_profile.3.md

```diff
@@ -0,0 +1,260 @@
+
+# SYNOPSIS
+
+```c
+include <rdma/fi_profile.h>
+
+int fi_profile_open(struct fid *fid, uint64_t flags,
+    struct fid_profile **prof_fid, void *context);
+
+int fi_profile_close(struct fid_profile *prof_fid);
+
+void fi_profile_reset(struct fid_profile *prof_fid,  uint64_t flags);
+
+ssize_t fi_profile_query_vars(struct fid_profile *prof_fid,
+    struct fi_profile_desc *varlist, int *count)
+
+ssize_t fi_profile_query_events(struct fid_profile *prof_fid,
+    struct fi_profile_desc *eventlist, int *count);
+
+int fi_profile_register_callback(struct fid_profile *prof_fid, uint32_t event,
+    int (*callback)(struct fid_profile *prof_fid, uint32_t event,
+                    void *param, size_t size, void *context),
+    void *context);
+
+int fi_profile_read_u64(struct fid_profile *prof_fid, uint32_t var_id,
+    uint64_t *data);
+
+void fi_profile_lock(struct fid_profile *prof_fid);
+
+void fi_profile_unlock(struct fid_profile *prof_fid);
+
+```
+
+# ARGUMENTS
+
+*fid*
+: Reference to an endpoint or a domain which is the target for profiling.
+
+*prof_fid*
+: Reference to a profile object associating with the target fid.
+
+*flags*
+: Flags to open and/or control a profiling.
+
+*context*
+: Context associated with the profiling operations.
+
+*varlist*
+: An array of profiling variables.
+
+*eventlist*
+: An array of profiling events.
+
+*count*
+: Number of variables or events in the array.
+
+*var_id*
+: An unqie ID for a variable.
+
+*event*
+: A unqie ID for an event. 
+
+*callback*
+: Callback function registered for an event.
+
+*param*
+: The parameter of the callback function.
+
+*size*
+: The amount of the data the provider is provided in the "param" 
+when the callback is called.
+
+*data*
+: An uint64_t value of a variable.
+
+# DESCRIPTION
+
+Fabric profiling interface allows users to obtain some of configuration or
+runtime properties from the providers in the form of "variables". Providers 
+define variables and make them readable to users through the profiling 
+interface. Providers define events and use them to report an operation 
+occured or a state changed where one or more variables may change their 
+values. Users can register callback functions to one or more events, and 
+receive notifications from providers once an event occurs.
+
+In order to support the profiling interface, an endpoing or a domain in a 
+provider needs to maintain a fid_profile object, and implement the profiling 
+operations there. To starts a profiling,  an open call must be made on the 
+target endpoint or domain object to obtain the fid_profile object.  If the 
+target object does not support profiling, the open call will fail with error 
+code -FI_ENOSYS. Once a fid_profile object is obtained successfully, it can be 
+used to query available variables or events, and read variables' values.
+
+# VARIABLES
+
+The profiling variables are defined in providers, and become accessable 
```

Comment:
wording/spelling: "are defined by providers and are accessible through the profiling interface.  Variables are specified using struct fi_profile_desc described below.  A variable can be a configuration..."